src/flatc.cpp

```diff
@@ -234,12 +241,33 @@ int main(int argc, const char *argv[]) {
         printf("flatc version %s\n", FLATC_VERSION);
         exit(0);
       } else {
+        std::map<int, int> gen_to_grpc_index;
+        gen_to_grpc_index[2] = 9;
+        gen_to_grpc_index[7] = 10;
         for (size_t i = 0; i < num_generators; ++i) {
           if (arg == generators[i].generator_opt_long ||
               (generators[i].generator_opt_short &&
                arg == generators[i].generator_opt_short)) {
             generator_enabled[i] = true;
             any_generator = true;
+            if(is_grpc) {
+              // Since grpc generators start from index 9
+              // Reset the grpc generators
```

Comment:
We shouldn't do all this hard-coding. A better way would be to have an additional function pointer in the structures above (which may null), for the GRPC code generation.